src/firecracker/examples/uffd/uffd_utils.rs

```diff
@@ -99,23 +99,24 @@ impl UffdPfHandler {
         }
     }
 
-    fn populate_from_file(&self, region: &MemRegion) -> (u64, u64) {
-        let src = self.backing_buffer as u64 + region.mapping.offset;
-        let start_addr = region.mapping.base_host_virt_addr;
-        let len = region.mapping.size;
-        // Populate whole region from backing mem-file.
-        // This offers an example of how memory can be loaded in RAM,
-        // however this can be adjusted to accommodate use case needs.
+    fn populate_from_file(&self, region: &MemRegion, dst: u64) -> (u64, u64) {
+        let offset = dst - region.mapping.base_host_virt_addr;
+        let src = self.backing_buffer as u64 + region.mapping.offset + offset;
+
+        // Populate a single page from backing memory file.
+        // This is just an example, probably, with the worst-case latency scenario,
+        // of how memory can be loaded in guest RAM.
+        let len = get_page_size().unwrap();
```

Comment:
I understand this is an example, but doing a `sysconf` for each page fault to get a page size seems a bit too much. Maybe move page size outside of the function and pass it as a parameter?

Reply:
Fixed